File: 2320-count-number-of-ways-to-place-houses/2320-count-number-of-ways-to-place-houses.py
class Solution:
    def countHousePlacements(self, n: int) -> int:
        if n == 1: return 4
        if n == 2: return 9
        pprev, prev, mod = 2, 3, pow(10, 9) + 7
        for i in range (3, n + 1):
            current = (pprev + prev)
            pprev, prev = prev, current
        return (prev * prev) % mod

File: 2320-count-number-of-ways-to-place-houses/2320-count-number-of-ways-to-place-houses.java
class Solution {
    public int countHousePlacements(int n) {
        if(n == 1) return 4;
        if(n == 2) return 9;
        long pprev = 2, prev = 3, mod = 1000000007;
        for(int i = 3; i <= n; ++i) {
            long current = (prev + pprev) % (mod);
            pprev = prev;
            prev = current;
        }
        return (int) ((prev * prev) % (mod));
    }
}